add Dialog component

hcourthias opened the pull request:
## Checklist before requesting a review
- [x] Working on iOS
- [x] Working on Android
- [x] Integration tests added
- [x] Visual regressions screenshots are up to date
- [ ] Design validated

Commit:
add react-native-safe-area-context

Commit:
add dialog component

Commit:
add screenshots

Commit:
add ios screenshot

Commit:
add ios screenshot

hcourthias commented on packages/sandbox/e2e/ios/screenshots/Dialog.png:
@ianrobertlambert The margin around the modal will not be their when using the component in the main project.

Commit:
add tests

judastheo commented on packages/streamline/src/components/dialog/dialog.tsx:
```diff
@@ -0,0 +1,97 @@
+        flexDirection={
+          buttons.orientation === 'horizontal' ? 'row-reverse' : 'column'
+        }
+        style={{ marginBottom: insets.bottom }}
```
You can move this style and the `useSafeAreaInsets` in the `useStyles` part

judastheo commented on packages/streamline/src/components/dialog/dialog.tsx:
```diff
@@ -0,0 +1,97 @@
+      ...(buttons.secondary ? { marginBottom: spacing.xxs } : {}),
+    },
+    secondary: {
+      ...(buttons.orientation === 'horizontal'
```
You check 3 times in the file `buttons.orientation === 'horizontal'`, you could have only one `isHorizontal` const.

judastheo commented on packages/streamline/src/components/dialog/dialog.types.tsx:
```diff
@@ -0,0 +1,16 @@
+      appearance: 'primary' | 'danger' | 'warning';
+    };
+    secondary?: ButtonProps & {
+      appearance: 'neutral' | 'secondary';
```
@hcourthias @ianrobertlambert Better safe than sorry, make sure you're aligned on possible appearance for each button.

judastheo commented on packages/streamline/src/primitives/icon/icon.constants.ts:
```diff
@@ -2,10 +2,12 @@ export const ICON_SIZE = {
   small: 16,
   regular: 24,
   large: 48,
+  xlarge: 48,
```
Discussed with @hcourthias , we may change it later but for now, it's just to match the icon different possibilities but should be of no use.

judastheo reviewed (changes requested):
Nice!

Commit:
Theo feedbacks
- create a isHorizontal const
- move insets from useSafeAreaInsets inside the useStyles

Commit:
fix margin

Commit:
update screenshots

ianrobertlambert commented:
Thanks @hcourthias - It's looking good! ✨ I have some questions and small feedback hopefully:

- [ ] The text content should be centered when the dialog uses an an icon.
- [ ] Does the `Bold` typography use a different typeface? I noticed that it was different on the screenshots, but not on the Expo environment? Is this issue specific to the `Dialog` component or is it a general `Typography` issue?
- [ ] Is there a way to test these in an overlay setting? It might provide a more accurate representation of their behavior on iOS/Android.
- [ ] Just to confirm, is the button(s) positioned 16px away from the edge of the `HomeIndicator` container on iOS/Android?
![image](https://github.com/GetLuko/streamline/assets/34472045/6eb4df60-481c-44f8-894e-24a0af48954b)
